.gitlab/includes/nvhpc24_9_pipeline.yml

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2023 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file BOOST_LICENSE_1_0.rst or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+include:
+  - local: '.gitlab/includes/common_pipeline.yml'
+  - local: '.gitlab/includes/common_spack_pipeline.yml'
+
+.variables_nvhpc24_9_config:
+  variables:
+    SPACK_ARCH: linux-ubuntu22.04-neoverse_v2
+    COMPILER: gcc@14.2.0
+    NVHPC_COMPILER: nvhpc@24.9
+    CXXSTD: 20
+    GPU_TARGET: '60'
+    SPACK_SPEC: "pika@main arch=$SPACK_ARCH %${NVHPC_COMPILER} +stdexec +mpi +cuda cuda_arch=${GPU_TARGET} \
+                 malloc=system cxxstd=$CXXSTD ^boost@1.78.0%${NVHPC_COMPILER} ^hwloc@2.7.0 \
+                 ^stdexec@24.09 ^cuda@12.4%${NVHPC_COMPILER} ^whip%${NVHPC_COMPILER}"
+    CMAKE_FLAGS: "-DPIKA_WITH_CXX_STANDARD=$CXXSTD -DPIKA_WITH_CUDA=ON -DPIKA_WITH_MALLOC=system \
+                  -DPIKA_WITH_MPI=ON -DCMAKE_CUDA_ARCHITECTURES=$GPU_TARGET -DPIKA_WITH_STDEXEC=ON"
+    PULL_ON_HOST_MACHINE: "YES"
+
+nvhpc24_9_spack_compiler_image:
+  extends:
+    - .variables_nvhpc24_9_config
+    - .compiler_image_template_santis
+
+nvhpc24_9_spack_image:
+  needs: [nvhpc24_9_spack_compiler_image]
+  extends:
+    - .variables_nvhpc24_9_config
+    - .dependencies_image_template_santis
+
+nvhpc24_9_build:
+  timeout: 2 hours
+  needs: [nvhpc24_9_spack_image]
+  extends:
+    - .variables_nvhpc24_9_config
+    - .build_template_santis
+
+#TODO add a test step once ci on alps works well
```

Comment:
I think you can remove this comment. This will be covered by #1239 and probably some other follow-up PRs.